### PR TITLE
OH2-395 | Add endpoint to update user group permissions accepting list of permissions ids

### PR DESCRIPTION
**Pull request by SteveGT96**
The new endpoints `PUT /usergroups/{group_code}/permissions` and  `PATCH /usergroups/{group_code}/permissions` accept a list of permissions ids(through the UserGroupPermissionsDTO).
For both endpoints:

-  Ids corresponding to permissions already assigned to the group will be skipped.
- ids that corresponding permission are not yet assigned to the groups will be assigned.
- Permissions ids that don't exist are ignored
- The full list of group permissions(PermissionDTO) is return.

For the patch endpoint:
- User group permissions that don't have their id in the permission ids payload will be removed
- That mean the PATCH endpoint just replace the group permissions with the ones provided in the payload.

See [OH2-395](https://openhospital.atlassian.net/browse/OH2-395)
Paired with [#1426](https://github.com/informatici/openhospital-core/pull/1426)

[OH2-395]: https://openhospital.atlassian.net/browse/OH2-395?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/org/isf/usergroups/dto/GroupPermissionsDTO.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package org.isf.usergroups.dto;
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/usergroups/dto/GroupPermissionsDTO.java:22]: where is the license header?

File: src/main/java/org/isf/usergroups/rest/UserGroupController.java [hunk 2]
```diff
@@ -214,11 +220,70 @@ public int assignPermission(
...
+	 */
+	@PatchMapping(value = "/usergroups/{group_code}/permissions", produces = MediaType.APPLICATION_JSON_VALUE)
+	public List<PermissionDTO> replaceGroupPermissions(
+		@PathVariable("group_code") String userGroupCode,
+		@RequestBody GroupPermissionsDTO payload
+	) throws OHServiceException {
+		LOGGER.info("Attempting to replace user group({}) permissions, with permissions ids, {}", userGroupCode, payload.permissions());
+		UserGroup userGroup = userManager.findUserGroupByCode(userGroupCode);
+		if (userGroup == null) {
+			LOGGER.info("Could not find user corresponding to the group code {}", userGroupCode);
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/usergroups/rest/UserGroupController.java]:
```suggestion
			LOGGER.info("Could not find user group corresponding to the group code {}", userGroupCode);
```

File: src/main/java/org/isf/usergroups/dto/GroupPermissionsDTO.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+package org.isf.usergroups.dto;
+
+import java.util.List;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+
+public record GroupPermissionsDTO(
+	@Schema(name = "permissions", example = "[48, 24]", description = "List of permissions' ids") List<Integer> permissions) {
```
Comment (SilverD3) [inline src/main/java/org/isf/usergroups/dto/GroupPermissionsDTO.java]:
```suggestion
	@Schema(name = "permissions", example = "[48, 24]", description = "List of permissions' ids") List<Integer> permissionIds) {

```
To be more precise.

File: .ide-settings/idea/OpenHospital-code-style-configuration.xml
Comment (SilverD3) [inline .ide-settings/idea/OpenHospital-code-style-configuration.xml:1]: I can't see what has changed in this file

**Review by SilverD3 — CHANGES_REQUESTED** [review]

**Comment by SilverD3** [comment]
Checks are not passing.

File: src/main/java/org/isf/usergroups/dto/GroupPermissionsDTO.java (hunk 3 above)
Comment (SteveGT96) [reply src/main/java/org/isf/usergroups/dto/GroupPermissionsDTO.java]: It's enough precise with type  and the description but if you want...

File: .ide-settings/idea/OpenHospital-code-style-configuration.xml
Comment (SteveGT96) [reply .ide-settings/idea/OpenHospital-code-style-configuration.xml:1]: Continuation indent.

**Comment by mwithi** [comment]
> Checks are not passing.

looks like is about new tests, @SteveGT96 could check that please?

**Comment by SteveGT96** [comment]
> > Checks are not passing.
> 
> looks like is about new tests, @SteveGT96 could check that please?

OK ! But it's strange 'cause UserGroupController Tests are passing.

**Comment by mwithi** [comment]
> OK ! But it's strange 'cause UserGroupController Tests are passing.

Yes, the problem is that [#1426](https://github.com/informatici/openhospital-core/pull/1426) is not yet merged

**Review by mwithi — APPROVED** [review]
LGTM. Thanks a lot!

**Pull request merged**